plugin/src/main/cpp/extensions/openxr_fb_environment_depth_extension_wrapper.cpp

```diff
@@ -0,0 +1,625 @@
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		return false;
+	}
+
+	XrEnvironmentDepthSwapchainCreateInfoMETA swapchain_create_info = {
+		XR_TYPE_ENVIRONMENT_DEPTH_SWAPCHAIN_CREATE_INFO_META, // type
+		nullptr, // next
+		0, // createFlags
+	};
+
+	result = xrCreateEnvironmentDepthSwapchainMETA(provider, &swapchain_create_info, &swapchain);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrCreateEnvironmentDepthSwapchainMETA failed!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	XrEnvironmentDepthSwapchainStateMETA swapchain_state = {
+		XR_TYPE_ENVIRONMENT_DEPTH_SWAPCHAIN_STATE_META, // type
+		nullptr, // next
+	};
+
+	result = xrGetEnvironmentDepthSwapchainStateMETA(swapchain, &swapchain_state);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrGetEnvironmentDepthSwapchainStateMETA failed!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	uint32_t image_count;
+	result = xrEnumerateEnvironmentDepthSwapchainImagesMETA(swapchain, 0, &image_count, nullptr);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrEnumerateEnvironmentDepthSwapchainImagesMETA failed to get count!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	struct XrSwapchainImageOpenGLKHR {
+		XrStructureType       type;
+		void* XR_MAY_ALIAS    next;
+		uint32_t              image;
+	};
+
+	Vector<XrSwapchainImageOpenGLKHR> swapchain_images;
+	for (int i = 0; i < image_count; i++) {
+		swapchain_images.push_back(XrSwapchainImageOpenGLKHR {
+			XR_TYPE_SWAPCHAIN_IMAGE_OPENGL_KHR, // type
+			nullptr, // next
+			0, // image
+		});
+	}
+
+	result = xrEnumerateEnvironmentDepthSwapchainImagesMETA(swapchain, image_count, &image_count, (XrSwapchainImageBaseHeader *)&swapchain_images[0]);
+	if (XR_FAILED(result)) {
+		UtilityFunctions::print("xrEnumerateEnvironmentDepthSwapchainImagesMETA failed to get data!");
+		UtilityFunctions::print(get_openxr_api()->get_error_string(result));
+		xrDestroyEnvironmentDepthSwapchainMETA(swapchain);
+		swapchain = nullptr;
+		xrDestroyEnvironmentDepthProviderMETA(provider);
+		provider = nullptr;
+		return false;
+	}
+
+	swapchain_image_rids.clear();
+	for (int i = 0; i < image_count; i++) {
+		swapchain_image_rids.push_back(
+			RenderingServer::get_singleton()->texture_create_from_native_handle(
+				RenderingServer::TextureType::TEXTURE_TYPE_LAYERED,
+				Image::Format::FORMAT_RH, // GL_DEPTH_COMPONENT16 according to the docs
```

Comment:
Is this correct? The image is GL_DEPTH_COMPONENT16 (OpenGL) or VK_FORMAT_D16_UNORM (for Vulkan)